plugins/woocommerce/includes/class-wc-payment-gateways.php

```diff
@@ -113,6 +117,153 @@ public function init() {
 		}
 
 		ksort( $this->payment_gateways );
+
+		add_action( 'wc_payment_gateways_initialized', array( $this, 'on_payment_gateways_initialized' ), 10, 1 );
+		/**
+		 * Hook that is called when the payment gateways have been initialized.
+		 *
+		 * @since 8.5.0
+		 */
+		do_action( 'wc_payment_gateways_initialized' );
+	}
+
+	/**
+	 * Hook into payment gateway settings changes.
+	 *
+	 * @since 8.5.0
+	 */
+	public function on_payment_gateways_initialized() {
+		foreach ( $this->payment_gateways as $gateway ) {
+			$option_key = $gateway->get_option_key();
+			self::add_action( 'add_option_' . $option_key, array( $this, 'payment_gateway_settings_option_changed' ), 10, 2 );
+			self::add_action( 'update_option_' . $option_key, array( $this, 'payment_gateway_settings_option_changed' ), 10, 3 );
+		}
+	}
+
+	/**
+	 * Callback for when a gateway settings option was added or updated.
+	 *
+	 * @param mixed  $old_value Old value. Option name when called via add_option_ hook.
+	 * @param mixed  $value New value.
+	 * @param string $option Option name.
+	 * @since 8.5.0
+	 */
+	private function payment_gateway_settings_option_changed( $old_value, $value, $option = null ) {
+		if ( null === $option ) {
+			// We're in the add_option_ hook so there's no old value and parameter order is different.
+			$option    = $old_value;
+			$old_value = null;
+		}
+		if ( ! $this->was_gateway_enabled( $value, $old_value ) ) {
+			return;
+		}
+
+		// This is a change to a payment gateway's settings and it was just enabled. Let's send an email to the admin.
+		// "untitled" shouldn't happen, but just in case.
+		$this->notify_admin_payment_gateway_enabled( $value['title'] ?? 'untitled' );
+	}
+
+	/**
+	 * Email the site admin when a payment gateway has been enabled.
+	 *
+	 * @param string $gateway_title Gateway title.
+	 * @return bool Whether the email was sent or not.
+	 * @since 8.5.0
+	 */
+	private function notify_admin_payment_gateway_enabled( $gateway_title ) {
+		$admin_email          = get_option( 'admin_email' );
+		$user                 = get_user_by( 'email', $admin_email );
+		$username             = $user ? $user->user_login : $admin_email;
+		$gateway_settings_url = self_admin_url( 'admin.php?page=wc-settings&tab=checkout' );
+		$site_name            = wp_specialchars_decode( get_option( 'blogname' ), ENT_QUOTES );
+		$site_url             = home_url();
+		/**
+		 * Allows adding to the addresses that receive payment gateway enabled notifications.
+		 *
+		 * @param array $email_addresses Email addresses.
+		 * @since 8.5.0
+		 */
+		$email_addresses   = apply_filters( 'wc_payment_gateway_enabled_notification_email_addresses', array() );
```

Comment:
Not a required change, but I wonder if there would be value in making the gateway ID available here—for instance, if it was particularly important to also notify other people in the case of just one gateway?

(Though, we can also see if we get feedback to this effect and make a further iteration later only if needed.)

Reply:
Decided to pass in the whole gateway instance. 👍